### PR TITLE
Loading Content on the Go Feature and Spinner Feature

### PR DESCRIPTION
**Pull request by sanjitk7**
Search Feature, Uses a 'search object' passed as prop by the user to display 'number of matches' as badges on the chips.

In Advanced Mode, uses Redux Saga to use a callback passed to load content only whenever the title(chip) is clicked for the first time. On consequent clicks of already clicked titles the content is cached/stored from the first time. This enables the user to load content only when needed from an external API.

A Spinner is displayed during loading/waiting for content.

### CODE REVIEW

File: src/components/TitleSection.jsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React from "react";
+import { bindActionCreators } from "redux";
```
Comment (peswaram) [inline src/components/TitleSection.jsx:2]: This is unused, can you remove it?

File: src/components/TabSection.jsx [hunk 2]
```diff
@@ -140,6 +147,13 @@ class TabSection extends React.Component {
       );
     }
 
+    console.log("tabsection loading state:", this.props.contentLoading);
```
Comment (peswaram) [inline src/components/TabSection.jsx:150]: can this removed?

File: src/components/TitleSection.jsx (hunk 1 above)
Comment (sanjitk7) [reply src/components/TitleSection.jsx:2]: yes

File: src/components/TabSection.jsx (hunk 2 above)
Comment (sanjitk7) [reply src/components/TabSection.jsx:150]: yes on it

**Pull request merged**